### ISSUE TITLE
Adding Copy Number changes in the REVOLVER input

### ISSUE THREAD

**Issue by sunandinisharma** [zanmer/Mesothelioma_evolution_deciphering_drugable_somatic_alterations#3]
Hi,

I wanted to ask if you could explain a little how were the Copy number aberrations added as an input to REVOLVER? It will be great if I can get some more clarification on a method to follow to assign CCF and cluster to copy number changes.

Thanks

**Comment by zanmer** [comment]
First, you should join the overlapped copy number segments among multiple samples. I used GenomicRanges R package or HTSeq python package to do this. And you will generate a matrix of copy number segments where the value means present or absent of the copy number aberrations. And then you should annotate each segments with cytoband name and known driver genes according to the chromosome coordination. SCNAs were manually assigned to CCF clusters based on p/a pattern of each segments in samples.

**Comment by sunandinisharma** [comment]
Hi Kevin,
Thanks for a quick response. As you said, I have a  matrix with SCNA: 17p, 1p, 6q with presence and absence of these aberrations across all patient samples. Now, I am just stuck with how to assign these SCNA to the CCF cluster? I ran pyclone sample by sample meaning one at a time since these samples are not series samples but rather diagnostic. Pyclone gave ccf clusters with different mutations for each .tsv that I ran through pyclone. I need a hint how will I assign the cluster and determine the ccf for SCNA?

I will really appreciate your help. This will be very helpful for the completion of my phd deader station project.

Thanks,
Sunandini Sharma

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by zanmer** [comment]
You should run pyclone patient by patient, not sample by sample, means multiple samples from one patient are incorporated together to do pyclone analysis. And you should do "force calling" procedure, usually using [bam-readcount](https://github.com/genome/bam-readcount), to obtain the read counts supporting ref and alt alleles for the mutations not detected in individual sample. Also accurate ploidies and purities of tumor samples are necessary. When finish pyclone, you can construct a CCF matrix which row represent clusters and column for samples. 
cluster   s1        s2       s3       s4
1            0.97     0.93    0.96    0.92
2            0          0.70     0          0
3            0.48     0         0          0

According to the pattern of CCF values, you can manually assign SCNA to the corresponding clusters.

**Comment by sunandinisharma** [comment]
Thank you so much Kevin. I ran Pyclone  yesterday on the patient samples. This explanation given by you makes sense. Of note, the design of the experiment is such, I don't have multiple samples for a patient meaning this is not a multi region sequencing neither we have relapse biopsy. So, let's say Cluster 1 in patient 1 is not same as Cluster 1 in patient 2? Is that correct?
Please let me know if I am thinking in the right direction or I can share the CCF matrix that I will construct from the results. Maybe it will make more sense talking about this.

You have been a great help. I sincerely appreciate it a lot.

Thanks and kindest regards,
Sunandini

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by zanmer** [comment]
Yes, you are right. Clusters from different patients may share the same somatic events, but they are truly different tumor clones or cells. Pyclone clusters are only comparable among samples from the same patient . Besides, REVOVLER was designed to infer repeated evolutionary trajectories based on multi-region sequencing, not relapse datasets.
